src/python/pants/engine/target.py

```diff
@@ -1049,6 +1048,11 @@ def validate(self) -> None:
                 "`TargetGenerator.moved_field`s, to avoid redundant graph edges."
             )
 
+    @classmethod
+    def register_plugin_field(cls, field: Type[Field]) -> UnionRule:
+        cls.moved_fields = cls.moved_fields + (field,)
```

Comment:
@Eric-Arellano: I removed `@final` from the base implementation of `register_plugin_field` so that I could override it here and add the field to `moved_fields` when dealing with a target generator situation.

Please advise what the best way to handle plugin fields and target generators is.

Reply:
Also should the API allow choosing between `moved_fields` and `copied_fields`?

Reply:
> Also should the API allow choosing between `moved_fields` and `copied_fields`?

It will need to I think... otherwise this is an API change for any callers who were already using this method.

Reply:
I have added a `bool` parameter for now that controls whether to copy the field or not. It defaults to having the field be moved instead of copied.